example.go

```diff
@@ -0,0 +1,376 @@
+	}
+
+	// Print topics details.
+	fmt.Printf("\tname: \t\t\t%s\n", *result.Name)
+
+	// The number of partitions.
+	fmt.Printf("\tno of partitions: \t%d\n", *result.Partitions)
+
+	// The number of replication factor.
+	fmt.Printf("\treplication factor: \t%d\n", *result.ReplicationFactor)
+
+	// // The value of config property 'retention.ms'.
+	fmt.Printf("\tretention (ms): \t%d\n", *result.RetentionMs)
+
+	// // The value of config property 'cleanup.policy'.
+	fmt.Printf("\tcleanup policy: \t%s\n", *result.CleanupPolicy)
+
+	// Configs *TopicConfigs
+	fmt.Printf("\ttopic configs: \t\t%+v\n", *result.Configs)
+
+	// The replia assignment of the topic.
+	// ReplicaAssignments []ReplicaAssignment
+	for _, assignment := range result.ReplicaAssignments {
+		fmt.Printf("\tassignment:  \t\tid:%d,  \tbrokers: %+v\n", assignment.ID, assignment.Brokers)
+	}
+	return nil
+} // func.end
+
+func createTopic(serviceAPI *adminrestv1.AdminrestV1) error {
+	// Construct an instance of the createTopicOptionsModel.
+	createTopicOptionsModel := new(adminrestv1.CreateTopicOptions)
+	createTopicOptionsModel.Name = core.StringPtr("test-topic")
+	createTopicOptionsModel.Partitions = core.Int64Ptr(int64(26))
+	createTopicOptionsModel.PartitionCount = core.Int64Ptr(int64(1))
+
+	// Create the Topic.
+	response, operationErr := serviceAPI.CreateTopic(createTopicOptionsModel)
+	if operationErr != nil {
+		return fmt.Errorf("Error Creating Topics: %s\n", operationErr.Error())
+	}
+
+	// Check the result.
+	if response.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("Error Creating Topic: status %d\n", response.StatusCode)
+	}
+
+	fmt.Printf("\tname: %s created\n", *createTopicOptionsModel.Name)
+
+	return nil
+} // func.end
+
+func deleteTopic(serviceAPI *adminrestv1.AdminrestV1) error {
+	// Construct an instance of the DeleteTopicOptions model
+	deleteTopicOptionsModel := new(adminrestv1.DeleteTopicOptions)
+	deleteTopicOptionsModel.TopicName = core.StringPtr("test-topic")
+	deleteTopicOptionsModel.Headers = map[string]string{"x-custom-header": "x-custom-value"}
+
+	// Delete Topic
+	response, operationErr := serviceAPI.DeleteTopic(deleteTopicOptionsModel)
+	if operationErr != nil {
+		return fmt.Errorf("Error Deleting Topic: %s\n", operationErr.Error())
+	}
+
+	// Check the result.
+	if response.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("Error Deleting Topic: status %d\n", response.StatusCode)
+	}
+
+	fmt.Printf("\tname: %s deleted\n", *deleteTopicOptionsModel.TopicName)
+	return nil
+} // func.end
+
+func updateTopicDetails(serviceAPI *adminrestv1.AdminrestV1) error {
+	// Construct an instance of the UpdateTopicOptions model
+	updateTopicOptionsModel := new(adminrestv1.UpdateTopicOptions)
+	updateTopicOptionsModel.TopicName = core.StringPtr("test-topic")
+	updateTopicOptionsModel.NewTotalPartitionCount = core.Int64Ptr(int64(6))
+	updateTopicOptionsModel.Headers = map[string]string{"x-custom-header": "x-custom-value"}
```

Comment:
Do we need this header?
Ditto in other methods below

Reply:
removed